server/src/main/java/com/cloud/template/HypervisorTemplateAdapter.java

```diff
@@ -585,6 +595,7 @@ public boolean delete(TemplateProfile profile) {
             if (iStores == null || iStores.size() == 0) {
                 // Mark template as Inactive.
                 template.setState(VirtualMachineTemplate.State.Inactive);
+                templateDao.remove(template.getId());
```

Comment:
There's a `templateDao` member in this class and a `_tmpltDao` in `TemplateAdapterBase`, and though I like the first name better, its presence is pollution/technical debt. I think you should use the one from the based class.

Reply:
Done @DaanHoogland